paper-server/src/main/java/org/bukkit/craftbukkit/structure/CraftStructure.java

```diff
@@ -97,7 +114,7 @@ public void place(RegionAccessor regionAccessor, BlockVector location, boolean i
         access.setDelegate(handle);
         access.setStructureTransformer(new CraftStructureTransformer(handle, new ChunkPos(pos), blockTransformers, entityTransformers));
 
-        this.structure.placeInWorld(access, pos, pos, definedstructureinfo, randomSource, 2);
+        this.structure.placeInWorld(access, pos, pos, definedstructureinfo, randomSource, 2 | (placementOptions.isStrict() ? 816 : 0));
```

Comment:
This should use the actual Block constants and/or reference the placeStructure method it's taken from

Reply:
Fixed!